### PR TITLE
Refactors medical stacks used on simple animals.

### PR DESCRIPTION
**Pull request by harmonyn**
Simple animals had an entirely separate and shitty piece of code that handled healing with medical stacks, and it overrode the medical.dm code. This fucked up everything as well as causing some mobs that shouldn't be able to be healed, to be able to. 

This PR adds a new var for simple animals, healable, that determines whether medical stacks work on them. It defaults to 1.

Current list of simple_animals that cannot be healed with medical stacks:

Drones. Fixes https://github.com/tgstation/-tg-station/issues/9760.
Constructs.
Shades.
Slimes.
Revenants.
Slaughter demons.
Morphs.
Hivebots.
Statues. (AS IF THEY NEED IT, FUCK)

MERGE CONFLICT FUCKING FIXED, FUCK

### CODE REVIEW

**Comment by RemieRichards** [comment]
this code is illogical.
M is a carbon. drones are not carbon. the ```if(!istype(M))``` should be working.

**Comment by bgobandit** [comment]
that part of the code wasn't mine. for whatever reason it seems it simply doesn't work -- simple animals aren't carbons, but you can use bruise packs/etc on them

my suspicion is it's supposed to be a sanity test

**Comment by RemieRichards** [comment]
1. Yes I'm aware
2. Yes it is a sanity check, checking the mob is of the right type, but it's failed.

Please try expanding the ```if(!istype(M))``` to ```if(!istype(M, /mob/living/carbon))``` and see if that works.

**Comment by Incoming5643** [comment]
There's handling later in the code for healing simple animals, which is intended. This check, if it ever worked, was obsoleted and apparently forgotten.

**Comment by bgobandit** [comment]
is that code somewhere other than medical.dm

**Comment by Incoming5643** [comment]
Yep, there's cruddy, non OOP code in https://github.com/tgstation/-tg-station/blob/909fcdc9a532f452817f8da85d4b67f46fe156dd/code/modules/mob/living/simple_animal/simple_animal.dm that handles healing for simple animals

**Comment by bgobandit** [comment]
WELL THAT'S WHY THIS IS FUCKED UP THEN, WHY WOULD SOMEONE EVEN DO THAT

brb refactoring this to not suck

File: code/game/objects/items/stacks/medical.dm [hunk 1]
```diff
@@ -25,8 +25,8 @@
 		user << "<span class='danger'>\The [M] is dead, you cannot help [t_him]!</span>"
 		return
 
-	if(!istype(M))
-		user << "<span class='danger'>You don't know how to apply \the [src] to [M]...</span>"
+	if(!istype(M, /mob/living/carbon) && !istype(M, /mob/living/simple_animal))
+		user << "<span class='danger'>You don't know how to apply \the [src] to [M]</span>..."
```
Comment (Cheridan) [inline code/game/objects/items/stacks/medical.dm]: Make this a period. We use ellipses to indicate an action that takes time to do.

Comment (Miauw62) [inline code/game/objects/items/stacks/medical.dm]: also, make sure your spans wrap around the entire message

**Comment by bgobandit** [comment]
k

**Pull request merged**